File: yukka/src/main/java/com/example/yukka/model/social/powiadomienie/PowiadomienieDTO.java
package com.example.yukka.model.social.powiadomienie;

import java.time.LocalDateTime;
import java.util.Set;

import org.springframework.beans.factory.annotation.Value;

import lombok.Builder;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;


@Getter
@Setter
@Builder
@ToString
// Tak naprawdę to jest DTO, ale nie chce mi się zmieniać nazwy
public class PowiadomienieDTO {
    private Long id;
    private String typ;
    private Boolean przeczytane;
    private String tytul;
    private String odnosnik;
    private String opis;
    private Set<String> nazwyRoslin;
    @Builder.Default
    @Value("${powiadomienia.obraz.default.name}")
    private String avatar = "default_powiadomienie_avatar.png";
    private String uzytkownikNazwa;
    private int iloscPolubien;
    private LocalDateTime data;
    @Builder.Default
    private LocalDateTime dataUtworzenia = LocalDateTime.now();

}

File: yukka-ui/src/app/services/models/powiadomienie-dto.ts
/* tslint:disable */
/* eslint-disable */
import { Uzytkownik } from '../models/uzytkownik';
export interface PowiadomienieDto {
  avatar?: string;
  data?: string;
  dataUtworzenia?: string;
  id?: number;
  iloscPolubien?: number;
  nazwyRoslin?: Array<string>;
  odnosnik?: string;
  okres?: string;
  opis?: string;
  przeczytane?: boolean;
  typ?: string;
  tytul?: string;
  uzytkownikNazwa?: string;
  zglaszajacy?: Uzytkownik;
  zglaszany?: string;
  zgloszenie?: boolean;
}
